js/modules/k6/ws/ws.go

```diff
@@ -293,6 +293,10 @@ func (*WS) Connect(ctx context.Context, url string, args ...goja.Value) (*WSHTTP
 				Value:  1,
 			})
 			socket.handleEvent("message", rt.ToValue(string(readData)))
+			if _, ok := socket.eventHandlers["binaryMessage"]; ok {
+				ab := rt.NewArrayBuffer(readData)
+				socket.handleEvent("binaryMessage", rt.ToValue(&ab))
+			}
```

Comment:
No matter whether it is 1 or 2 events I do think that we need to check the messagetype and call either the correct method or if we choose to have only `message` events change the type of the value to ArrayBuffer or string based on it.

This will require more work in `readPump` which I think should probably be better handled in a separate PR , possibly even in a following release so ... I am mostly for dropping the `ws` changes for now 

WDYT @na-- @imiric

Reply:
You're right, we should only call the `binaryMessage` handler if the `messageType == BinaryMessage`, I overlooked that.

It shouldn't be much work to refactor, I'll look into it today. Don't think we should delay this because of that change.

Reply:
@MStoykov Let me know if 5e29bdfb is what you had in mind. I'll fix the new linter errors, and probably rebase on `master`.